torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -310,7 +310,13 @@ void pushPackingPastRnn(Block* b) {
       std::vector<int64_t> new_sizes;
       new_sizes.push_back(*oldType->sizes()[0]);
       new_sizes.push_back(*oldType->sizes()[1]);
-      new_sizes.push_back(rnn->i(attr::hidden_size));
+      if (next->kind() == onnx::Reshape) {
```

Comment:
Interesting...is this fixing an existing bug?

Reply:
yes, the bug was undiscovered before because in unit tests, this output is graph output, so its shape assignment here is overwritten by the output tensor's shape. Now that shape inference is comparing inferred shape with output tensor's shape, this contradiction is discovered.